### PR TITLE
Normalise URNs

### PR DESCRIPTION
**Pull request by KaitCrawford**

### CODE REVIEW

File: casepro/utils/__init__.py [hunk 1]
```diff
@@ -80,6 +81,42 @@ def normalize(text):
...
+    # remove other characters
+    number = re.sub('[^0-9\+]', '', number.lower())
+
+    # add on a plus if it looks like it could be a fully qualified number
+    if len(number) >= 11 and number[0] not in ['+', '0']:
+        number = '+' + number
+
+    normalized = None
+    try:
+        normalized = phonenumbers.parse(number, None)
```
Comment (KaitCrawford) [inline casepro/utils/__init__.py]: The `None` here needs to be replaced with a country code for this to work.

**Comment by smn** [comment]
I don't know how valuable normalising msisdns without a country prefix, it's not really able to be done reliable without one.

**Comment by smn** [comment]
As discussed in Slack, the suggestion is adding the country code to the organisation, similar to how timezones are stored.

File: casepro/utils/__init__.py [hunk 2]
```diff
@@ -80,6 +81,42 @@ def normalize(text):
...
+    except Exception:
```
Comment (smn) [inline casepro/utils/__init__.py]: rather catch `phonenumbers.phonenumberutil.NumberParseException` here instead, `Exception` is too liberal to be useful.

File: casepro/utils/__init__.py [hunk 3]
```diff
@@ -80,6 +81,42 @@ def normalize(text):
...
+        pass
+
+    # now does it look plausible?
+    try:
+        if phonenumbers.is_possible_number(normalized):
+            return phonenumbers.format_number(normalized, phonenumbers.PhoneNumberFormat.E164), True
+    except Exception:
```
Comment (smn) [inline casepro/utils/__init__.py]: We may want to be bit more specific here as well in what we're trying to catch.

**Comment by jpaidoussi** [comment]
@rudigiesler Is this the direction you were thinking of?

**Comment by rudigiesler** [comment]
@jpaidoussi Yes, this looks good to me. We just have to think about how we're going to replicate this on the front end (maybe we could expose the urn check on an API endpoint?). :+1:

**Pull request merged**